File: dongyimai_manager_web/src/main/webapp/js/service/itemCatService.js
//商品类目服务层
app.service('itemCatService', function ($http) {

    //findAll
    this.findAll = function () {
        return $http.get('../itemCat/findAll.do');
    }

    //findPage
    this.findPage = function (page, rows) {
        return $http.get('../itemCat/findPage.do?page=' + page + '&rows=' + rows);
    }

    //findOne
    this.findOne = function (id) {
        return $http.get('../itemCat/findOne.do?id=' + id);
    }

    //add
    this.add = function (entity) {
        return $http.post('../itemCat/add.do', entity);
    }

    //update
    this.update = function (entity) {
        return $http.post('../itemCat/update.do', entity);
    }

    //dele
    this.dele = function (ids) {
        return $http.get('../itemCat/delete.do?ids=' + ids);
    }

    //条件、模糊查询+分页
    this.search = function (page, rows, searchEntity) {
        return $http.post('../itemCat/search.do?page=' + page + "&rows=" + rows, searchEntity);
    }

    //通过父级ID查询本级商品分类列表（下钻法，内关联表）
    this.findByParentId = function (parentId) {
        return $http.get('../itemCat/findByParentId.do?parentId=' + parentId);
    }

});

File: dongyimai_sellergoods_interface/src/main/java/com/offcn/sellergoods/service/ItemCatService.java
package com.offcn.sellergoods.service;

import com.offcn.entity.PageResult;
import com.offcn.pojo.TbItemCat;

import java.util.List;

public interface ItemCatService {

	//findAll
	public List<TbItemCat> findAll();
	
	//findPage
	public PageResult findPage(int pageNum, int pageSize);
	
	//add
	public void add(TbItemCat item_cat);
	
	//update
	public void update(TbItemCat item_cat);
	
	//findOne
	public TbItemCat findOne(Long id);
	
	//delete
	public void delete(Long[] ids);

	//条件、模糊查询+分页
	public PageResult findPage(TbItemCat item_cat, int pageNum, int pageSize);


	/**
	 * 通过父级ID查询本级SPU商品分类列表
	 * @param parentId	父级ID
	 * @return
	 */
	public List<TbItemCat> findByParentId(Long parentId);
	
}
